src/site/site.controller.ts

```diff
@@ -1,36 +1,62 @@
 import {
-  Controller, Get, Post, Body, Patch, Param, Delete
+  Controller, Get, Param, Delete, Query, Req, ParseIntPipe
 } from '@nestjs/common'
+import type { Request } from 'express'
 import { SiteService } from './site.service'
-import { CreateSiteDto } from './dto/create-site.dto'
-import { UpdateSiteDto } from './dto/update-site.dto'
+import { Site } from '.prisma/client'
+import { ItemsQueryPipe } from '~/api-services/pagination/pipes/ItemsQueryPipe'
+import { PageQueryPipe } from '~/api-services/pagination/pipes/PageQueryPipe'
+import { PaginatedData } from '~/api-services/pagination/dto/PaginationData'
+import { createPaginationData } from '~/api-services/pagination/creator/createPaginationData'
+import { Prisma } from '@prisma/client'
+import { ApiExtraModels, ApiTags } from '@nestjs/swagger'
+import { ApiPaginatedResponse } from '~/api-services/pagination/ApiPaginationResponse'
+import { GetSiteDto } from './dto/get-site.dto'
 
-@Controller('site')
+@Controller('sites')
+@ApiTags('sites')
+@ApiExtraModels(PaginatedData)
 export class SiteController {
   constructor(private readonly siteService: SiteService) {}
 
-  @Post()
-  create(@Body() createSiteDto: CreateSiteDto) {
-    return this.siteService.create(createSiteDto)
-  }
+  // @Post()
+  // create(@Body() createSiteDto: CreateSiteDto) {
+  //   return this.siteService.create(createSiteDto)
+  // }
 
   @Get()
-  findAll() {
-    return this.siteService.findAll()
+  @ApiPaginatedResponse(GetSiteDto)
+  async findAll(
+    @Req() req: Request,
+    @Query('items', ItemsQueryPipe) items: number,
+    @Query('page', PageQueryPipe) page: number,
+  ): Promise<PaginatedData<Site>> {
+    const sites = await this.siteService.findAll({
+      skip: page * items,
+      take: items
+    })
+    const totalItemCount = await this.siteService.count()
+
+    return createPaginationData<Site>({
+      url: `${req.protocol}://${req.get('Host')}${req.url}`,
```

Comment:
Cette logique pourrait être géré dans la fonction `createPaginationData` directement en lui passant `req`

Reply:
On est déjà dans un environnement full nestjs donc c'est pas un soucis 👌

Reply:
:white_check_mark: in a987a15